src/main/java/com/hida/ArkMinter.java

```diff
@@ -0,0 +1,251 @@
+     * Similar to the Mode settings in genIDauto() method. d:
+     *
+     * <p>
+     * DIGIT: {0123456789} Digit values only.</p>
+     * <p>
+     * LOWERCASE: {abc...xyz} Lowercase letters only. </p>
+     * <p>
+     * UPPERCASE: {ABC...XYZ} Uppercase letters only.</p>
+     * <p>
+     * MIXCASE: {abc...XYZ} Lowercase and Uppercase letters only. </p>
+     * <p>
+     * EXTENDED: {012...xyz} Digit values and Lowercase letters only. </p>
+     * <p>
+     * ALPHANUMERIC: {012...abc...XYZ} Digit values, Lowercase and Uppercase
+     * letters.</p>
+     *
+     * @param token: There are 2 different modes to choose from. RANDOM: Value
+     * by value, each value's order is randomized. SEQUENTIAL: Value by value,
+     * each value's order is sequenced.
+     * @return - A reference to a JsonObject that contains Json list of ids
+     */
+    @Override
+    public JsonObject genIdAuto(String token) {
+        String characters = (String) this.tokenMaps.get(token);
+
+        int charactersLength = characters.length();
+        for (int j = 0; j < this.AMOUNT; j++) {
+            StringBuilder buffer = new StringBuilder();
+            for (int i = 0; i < this.LENGTH; i++) {
+                double index = Math.random() * charactersLength;
+                buffer.append(characters.charAt((int) index));
+            }
+            String temp = buffer.toString();
+            this.ID_ARRAY.add(NAAN + this.PREFIX + temp);
+        }
+        return convertListToJson();
+    }
+
+    /**
+     * genIDcustom() Method description: This generates an customized ID format.
+     * User would give a string that excepts 6 letters: {d, l, u, m, e, a}.
+     * Similar to the Mode settings in genIDauto() method. d:
+     *
+     * <p>
+     * DIGIT: {0123456789} Digit values only.</p>
+     * <p>
+     * LOWERCASE: {abc...xyz} Lowercase letters only. </p>
+     * <p>
+     * UPPERCASE: {ABC...XYZ} Uppercase letters only.</p>
+     * <p>
+     * MIXCASE: {abc...XYZ} Lowercase and Uppercase letters only. </p>
+     * <p>
+     * EXTENDED: {012...xyz} Digit values and Lowercase letters only. </p>
+     * <p>
+     * ALPHANUMERIC: {012...abc...XYZ} Digit values, Lowercase and Uppercase
+     * letters.</p>
+     *
+     * @param isRandom - generates ids based on whether or not user requested
+     * random strings
+     * @return - A reference to a JsonObject that contains Json list of ids
+     */
+    @Override
+    public JsonObject genIdCustom(boolean isRandom) {
+        String format = this.FORMAT.toLowerCase();
+
+        double index = 0.0D;
+        for (int j = 0; j < this.AMOUNT; j++) {
+            StringBuilder buffer = new StringBuilder();
+            for (int i = 0; i < format.length(); i++) {
+                char c = format.charAt(i);
+                if (isRandom) {
+                    switch (c) {
+                        case 'd':
+                            index = Math.random() * 10.0D;

```

Comment:
Instead of having hard-coded constants for the length of the string, use `length()` instead